src/Polly.Core/ResilienceStrategyExtensions.Async.Task.cs

```diff
@@ -0,0 +1,72 @@
+using System.Threading;
+using Polly;
+
+namespace Polly;
+
+public static partial class ResilienceStrategyExtensions
+{
+    /// <summary>
+    /// Executes the <paramref name="execution"/> callback.
+    /// </summary>
+    /// <typeparam name="TState">The type of state associated with the execution.</typeparam>
+    /// <param name="strategy">Instance of <see cref="IResilienceStrategy"/>.</param>
+    /// <param name="execution">The execution callback.</param>
+    /// <param name="context">The context associated with the execution.</param>
+    /// <param name="state">The state associated with the execution.</param>
+    /// <returns>The instance of <see cref="Task"/> that represents the asynchronous execution.</returns>
+    public static async Task ExecuteAsTaskAsync<TState>(
+        this IResilienceStrategy strategy,
+        Func<ResilienceContext, TState, Task> execution,
+        ResilienceContext context,
+        TState state)
+    {
+        Guard.NotNull(strategy);
+        Guard.NotNull(execution);
+        Guard.NotNull(context);
+
+        InitializeAsyncContext(context);
+
+        await strategy.ExecuteInternalAsync(
+            static async (c, s) =>
```

Comment:
Could we use more meaningful names here than `c` and `s`? With this being a new API and me not looking at it in an IDE, it isn't super clear to me what the two things are (I'm assuming `context` and `strategy`?).

Reply:
Ah I see in the method after they're `(context, state)`.

Reply:
Done.